libs/vkd3d/vkd3d_private.h

```diff
@@ -4885,6 +4932,36 @@ UINT d3d12_determine_shading_rate_image_tile_size(struct d3d12_device *device);
 bool d3d12_device_supports_required_subgroup_size_for_stage(
         struct d3d12_device *device, VkShaderStageFlagBits stage);
 
+static inline void d3d12_device_register_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain)
+{
+    spinlock_acquire(&device->low_latency_swapchain_spinlock);
+
+    dxgi_vk_swap_chain_incref(chain);
+
+    if (!device->swapchain_info.low_latency_swapchain && device->swapchain_info.swapchain_count == 0)
+        device->swapchain_info.low_latency_swapchain = chain;
+    else
+        device->swapchain_info.low_latency_swapchain = NULL;
+
+    device->swapchain_info.swapchain_count++;
+
+    spinlock_release(&device->low_latency_swapchain_spinlock);
+}
+
+static inline void d3d12_device_remove_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain)
+{
+    spinlock_acquire(&device->low_latency_swapchain_spinlock);
+
+    dxgi_vk_swap_chain_decref(chain);
+
+    if (device->swapchain_info.low_latency_swapchain == chain)
```

Comment:
Similarly, we only need the decref when we're removing the low latency chain specifically.

Reply:
Done. I also added a decref when a second swapchain is added and the low_latency_swapchain is set to null.